public/app/features/alerting/unified/components/silences/SilencesTable.tsx

```diff
@@ -61,7 +65,7 @@ const SilencesTable: FC<Props> = ({ silences, alertManagerAlerts, alertManagerSo
       {!!silences.length && (
         <>
           <SilencesFilter />
-          {contextSrv.isEditor && (
+          {hasEditPermissions && (
```

Comment:
What do you think about creating a simple component called e.g. `<Authorize>` or <AccessControl> or something else which hides that condition logic from the jsx?
It might look like this:
```ts
<Authorize permission={[AccessControlAction.AlertingInstanceCreate, contextSrv.isEditor]}>
  <SecuredComponent />
</Authorize>
```

Reply:
I like this approach!